src/status_im/contexts/profile/edit/bio/events.cljs

```diff
@@ -1,22 +1,26 @@
 (ns status-im.contexts.profile.edit.bio.events
-  (:require [utils.i18n :as i18n]
+  (:require [clojure.string :as string]
+            [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
 (rf/reg-event-fx :profile/edit-profile-bio-success
- (fn [_]
+ (fn [_ [added?]]
    {:fx [[:dispatch [:navigate-back]]
          [:dispatch
           [:toasts/upsert
            {:type  :positive
             :theme :dark
-            :text  (i18n/label :t/bio-added)}]]]}))
+            :text  (if added?
+                     (i18n/label :t/bio-added)
+                     (i18n/label :t/bio-updated))}]]]}))
 
 (defn edit-profile-bio
-  [{:keys [db]} [bio]]
-  {:db (assoc-in db [:profile/profile :bio] bio)
-   :fx [[:json-rpc/call
-         [{:method     "wakuext_setBio"
-           :params     [bio]
-           :on-success [:profile/edit-profile-bio-success]}]]]})
+  [{:keys [db]} [new-bio]]
+  (let [{:keys [bio]} (:profile/profile db)]
+    {:db (assoc-in db [:profile/profile :bio] new-bio)
+     :fx [[:json-rpc/call
+           [{:method     "wakuext_setBio"
```

Comment:
@cammellos, should we start adding contract tests for all new rpc calls? It's rather simple :)

Reply:
Sure, makes sense, thanks for pointing this out, @mohsen-ghafouri would you mind creating a contract test for this? @J-Son89 can point you in the right direction

Reply:
for sure, 
I added the first set of tests in 
https://github.com/status-im/status-mobile/pull/18665

please see the wallet tests - https://github.com/status-im/status-mobile/pull/18665/files#diff-a5de5130c6b8a61e01a9fc957ec5546595fcee043de147bd890a732183b45587

you can probably just add a similar ns, like `tests.contract-test.profile-test`

to run the tests, you can run `make test-contract`.

In these test we want to test the rpc requests that we make to status go, but please do so for only the use cases that we use, we don't want to cover cases beyond our use cases 👌 

Any further questions, feel to reach out. Probably we will need to add more documentation etc, any and all feedback will be welcome there too

Reply:
is it like integration test that i added here? what's the difference? @J-Son89 
https://github.com/status-im/status-mobile/blob/9362d12bbddea497f4efc943f0510d3194e1d09d/src/tests/integration_test/profile_test.cljs#L60

Reply:
Added @J-Son89, could you please check again?